controllers/routemonitor/routemonitor_supplement.go

```diff
@@ -235,7 +236,24 @@ func (r *RouteMonitorReconciler) EnsureRouteURLExists(route routev1.Route, route
 	}
 
 	if currentRouteURL != "" && extractedRouteURL != currentRouteURL {
-		r.Log.V(3).Info("RouteURL mismatch: currentRouteURL and extractedRouteURL are not equal, taking extractedRouteURL as source of truth")
+		client := &http.Client{
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		}
+
+		resp, err := client.Head(extractedRouteURL)
+		if err != nil {
+			r.Log.V(3).Error(err, "Failed to make HEAD request to the URL")
+		}
+
+		if resp != nil && (resp.StatusCode == http.StatusMovedPermanently || resp.StatusCode == http.StatusFound || resp.StatusCode == http.StatusSeeOther || resp.StatusCode == http.StatusTemporaryRedirect || resp.StatusCode == http.StatusPermanentRedirect) {
```

Comment:
Would it be easier we just check the return code matches regex `3xx`?

Reply:
My understanding is 300 and 304 aren't exact fit for redirection. https://pkg.go.dev/net/http#Client.Get